### PR TITLE
Fixes for activities endpoints.

### PR DESCRIPTION
**Pull request by svogt0511**
## Purpose

Should fix a several problems.  

1. The first is the https://github.com/datacite/lupo/issues/1054, where trying to retrieve activities for a DOI would cause a 'NoMethodError: undefined method 'name' for nil:NilClass' error when trying to display activities using /dois/:id/activities endpoint older DOIs in production.  This was a problem in the serializer.
2. Added a fix for https://github.com/datacite/lupo/issues/1060  where the /activities/:uid endpoint would always return not found.  Turns out it was not implemented. 
3. This is the same as item 1, described, above:  https://github.com/datacite/lupo/issues/1062
4. Minor fix for the dev environment setup so activities can successfully be tested manually: https://github.com/datacite/lupo/issues/1065
5. Also found it necessary in the activities serializer to guard to guard against audited changes that show an initial value of null for publisher or publisher_obj.  This happens when these fields are first assigned a value.  An example be found in the audit date for https://api.datacite.org/activities/7b333de3-77af-4200-8ffd-c8b2012a8d9e:

image

closes: https://github.com/datacite/lupo/issues/1054, https://github.com/datacite/lupo/issues/1060, https://github.com/datacite/lupo/issues/1062, https://github.com/datacite/lupo/issues/1065

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

- [ ] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Reviewer, please remember our [guidelines](https://datacite.atlassian.net/wiki/spaces/TEC/pages/1168375809/Pull+Request+Guidelines):

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Offer suggestions on how to improve code e.g. simplification or expanding clarity.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

File: app/controllers/activities_controller.rb [hunk 1]
```diff
@@ -145,8 +145,9 @@ def show
 
   protected
     def set_activity
-      response = Activity.find_by_id(params[:id])
-      @activity = response.results.first
+      response = Activity.find_by_request_uuid!(params[:id])
+      @activity = response
+      puts response.inspect
```
Comment (digitaldogsbody) [inline app/controllers/activities_controller.rb:150]: Is this line left over from debugging? If so it might be worth removing it before merge :)

**Review by digitaldogsbody — APPROVED** [review]
A few minor things but non-blocking

File: app/controllers/activities_controller.rb (hunk 1 above)
Comment (svogt0511) [reply app/controllers/activities_controller.rb:150]: oops, yes.

**Review by digitaldogsbody — APPROVED** [review]

**Review by ashwinisukale — APPROVED** [review]
Changes look good, just suggested some refactoring changes

File: app/controllers/activities_controller.rb [hunk 2]
```diff
@@ -145,8 +145,9 @@ def show
 
   protected
     def set_activity
-      response = Activity.find_by_id(params[:id])
-      @activity = response.results.first
+      response = Activity.find_by_request_uuid!(params[:id])
+      @activity = response
+
```
[the remark this answers is not in the document]
Comment (svogt0511) [reply app/controllers/activities_controller.rb:150]: Getting rid of the variable.  Thanks.  Didn't mean to leave that.

File: app/serializers/activity_serializer.rb [hunk 3]
```diff
@@ -14,55 +14,45 @@ class ActivitySerializer
              :version
 
   attribute :changes do |object, params|
```
[the remark this answers is not in the document]
Comment (svogt0511) [reply app/serializers/activity_serializer.rb:16]: The comments really clarify.  Otherwise I am just staying with what I have.  Thanks!

**Pull request merged**